sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -91,15 +90,20 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
   const CG::CGTYPE Type = CommandGroup->getType();
   std::vector<Command *> AuxiliaryCmds;
   std::vector<StreamImplPtr> Streams;
+  std::vector<std::shared_ptr<const void>> AuxiliaryResources;
 
   if (Type == CG::Kernel) {
     Streams = ((CGExecKernel *)CommandGroup.get())->getStreams();
+    ((CGExecKernel *)CommandGroup.get())->clearStreams();
```

Comment:
Can we outline `((CGExecKernel *)CommandGroup.get())` into a temporary variable? Or is it modified at each statement?

Also, we should probably prefer to use `static_cast` here over c-style cast.